src/Ryujinx.HLE/HOS/Services/ServerBase.cs

```diff
@@ -394,24 +486,31 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
-                foreach (IpcService service in _sessions.Values)
+                if (!_threadStopped.Wait(ThreadStoppedWaitTimeout))
                 {
-                    if (service is IDisposable disposableObj)
+                    Logger.Warning?.Print(LogClass.Service, $"the ServerBase thread didn't signal it has stopped within {ThreadStoppedWaitTimeout:g}, resources may be leaked!");
```

Comment:
```suggestion
                    Logger.Warning?.Print(LogClass.Service, $"The ServerBase thread didn't signal it has stopped within {ThreadStoppedWaitTimeout:g}, resources may be leaked!");
```

Also, it's not really "may be leaked" if you're skipping the clear and dispose of everything below.